pkg/deploy/rollback/rest.go

```diff
@@ -0,0 +1,62 @@
+package rollback
+
+import (
+	"errors"
+	"fmt"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kerrors "github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/deploy/api/validation"
+)
+
+type REST struct {
+	Generator rollbackGenerator
+}
+
+type rollbackGenerator interface {
+	Generate(namespace string, rollback *deployapi.DeploymentConfigRollback) (*deployapi.DeploymentConfig, error)
+}
+
+func NewREST(generator rollbackGenerator) apiserver.RESTStorage {
+	return &REST{Generator: generator}
+}
+
+func (s *REST) New() runtime.Object {
+	return &deployapi.DeploymentConfigRollback{}
+}
+
+func (s *REST) List(ctx kapi.Context, labels, fields labels.Selector) (runtime.Object, error) {
+	return nil, errors.New("deploy/rollback.REST.List() is not implemented")
+}
+
+func (s *REST) Get(ctx kapi.Context, id string) (runtime.Object, error) {
+	return nil, errors.New("deploy/rollback.REST.Get() is not implemented")
+}
+
+func (s *REST) Delete(ctx kapi.Context, id string) (<-chan apiserver.RESTResult, error) {
+	return nil, errors.New("deploy/rollback.REST.Delete() is not implemented")
+}
+
+func (s *REST) Update(ctx kapi.Context, obj runtime.Object) (<-chan apiserver.RESTResult, error) {
+	return nil, errors.New("deploy/rollback.REST.Update() is not implemented")
+}
+
+func (s *REST) Create(ctx kapi.Context, obj runtime.Object) (<-chan apiserver.RESTResult, error) {
+	rollback, ok := obj.(*deployapi.DeploymentConfigRollback)
+	if !ok {
+		return nil, fmt.Errorf("not a rollback spec: %#v", obj)
+	}
+
+	if errs := validation.ValidateDeploymentConfigRollback(rollback); len(errs) > 0 {
+		return nil, kerrors.NewInvalid("deploymentConfigRollback", "", errs)
+	}
+
+	return apiserver.MakeAsync(func() (runtime.Object, error) {
```

Comment:
Why does this method not retrieve both the rc and ref mentioned in the rbc and pass them to generate?  I don't see a reason that that should be abstracted from this class which has access to etcd.